addons/mail/tests/test_mail_activity.py

```diff
@@ -18,3 +19,69 @@ def test_chatter_activity_tour(self):
             "mail_activity_schedule_from_chatter",
             login="admin",
         )
+
+
+class TestActivityScheduleCommon(HttpCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user_admin = cls.env.ref('base.user_admin')
+        cls.model_res_partner = cls.env.ref('base.model_res_partner')
+        cls.partner_1, cls.partner_2, cls.partner_3 = cls.env['res.partner'].create([
+            {'name': f'partner{idx + 1}'} for idx in range(3)])
+        cls.activity_type_todo = cls.env.ref('mail.mail_activity_data_todo')
+        cls.activity_type_call = cls.env.ref('mail.mail_activity_data_call')
+        cls.plan_party = cls.env['mail.activity.plan'].create({
+            'name': 'Test Plan A Party',
+            'res_model_ids': False,
+            'template_ids': [Command.create({
+                'activity_type_id': cls.activity_type_todo.id,
+                'summary': 'Book a place',
+                'responsible': 'on_demand',
+            }), Command.create({
+                'activity_type_id': cls.activity_type_todo.id,
+                'summary': 'Invite special guest',
+                'responsible': 'other',
+                'responsible_id': cls.user_admin.id,
+            }),
+            ],
+        })
+        cls.plan_onboarding = cls.env['mail.activity.plan'].create({
+            'name': 'Test Onboarding',
+            'res_model_ids': False,
+            'template_ids': [Command.create({
+                'activity_type_id': cls.activity_type_todo.id,
+                'summary': 'Plan training',
+                'responsible': 'other',
+                'responsible_id': cls.user_admin.id,
+            }), Command.create({
+                'activity_type_id': cls.activity_type_todo.id,
+                'summary': 'Training',
+                'responsible': 'other',
+                'responsible_id': cls.user_admin.id,
+            }),
+            ]
+        })
+
+    def reverse_record_set(self, records):
+        """ Get an equivalent recordset but with elements in reversed order. """
+        return self.env[records._name].browse([record.id for record in reversed(records)])
+
+    def get_last_activities(self, on_record, limit=None):
+        """ Get the last activities on the record in id asc order. """
+        return self.reverse_record_set(self.env['mail.activity'].search(
+            [('res_model', '=', on_record._name), ('res_id', '=', on_record.id)], order='id desc', limit=limit))
+
+    def get_last_message(self, on_record):
+        """ Get the last non-internal message posted on the given record. """
+        return self.env['mail.message'].search([('model', '=', on_record._name),
+                                                ('is_internal', '=', False),
+                                                ('res_id', '=', on_record.id)], order='id desc', limit=1)
+
+    def new_activity_schedule_wizard_form(self, records, additional_context_value=None):
```

Comment:
Would make it private (even if test tool, always a good pratice), like
```suggestion
    def _instantiate_activity_schedule_wizard(self, records, additional_context_value=None):
```